### DISCUSSION TITLE
Problem with prestashop global variable

### DISCUSSION THREAD

**Discussion by adamczestkowski** [PrestaShop/PrestaShop#33688] [Q&A]
Hello, has anyone had a problem like the one in the screenshot? When I started debugging frontcontroller.php, the correct JSON was assigned to the 'prestashop' variable (im dump $templateVars in assignGeneralPurposeVariables()). What could have been the cause of this behavior?

![image](https://github.com/PrestaShop/PrestaShop/assets/3323372/6fa9a502-b57a-4eef-afd1-32767a0a8801)

PHP: 7.4.30
PrestaShop: 1.7.8.3

**Comment by kpodemski** [comment]
I've never seen anything like this 🤔 Are you sure the correct JSON is being assigned? Maybe there are some special characters and a problem with passing them to the actual view?

**Reply by adamczestkowski** [reply]
The string being assigned was empty, as if the buildFrontEndObject method was not returning anything. I'm not sure if the parameter passed to this method had any value. Is it possible that the system cache went haywire?
